CRM/Contribute/Form/UpdateSubscription.php

```diff
@@ -81,8 +81,10 @@ public function preProcess() {
 
     $this->contributionRecurID = CRM_Utils_Request::retrieve('crid', 'Integer', $this, FALSE);
     if ($this->contributionRecurID) {
-      $this->_paymentProcessor = CRM_Contribute_BAO_ContributionRecur::getPaymentProcessor($this->contributionRecurID);
-      if (!$this->_paymentProcessor) {
+      try {
+        $this->_paymentProcessor = CRM_Financial_BAO_PaymentProcessor::getPaymentProcessorForRecurringContribution($this->contributionRecurID);
```

Comment:
actually it needs to be $this->_paymentProcessorObj = CRM_Financial_BAO_PaymentProcessor::getPaymentProcessorForRecurringContribution($this->contributionRecurID);

I think, but the issue is a bit more than this. I think it's about making it so a cancellation happens if UI changes (changing installment number, end_date) effectively cancel it. I think it might be easier to change installments on the form when end_date changes - or force user to & possibly move this to the ContributionRecur BAO

Reply:
Agree of what need to be addressed, I made the revert to fix a issue mentioned by reporter - https://issues.civicrm.org/jira/browse/CRM-20800?focusedCommentId=108162&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-108162

 I though the same but then ```$this->_paymentProcessor``` will be set to NULL which later assigned at L140. Although I found no use of ```$this->_paymentProcessor``` anywhere in ```UpdateSubscription``` php or template, so we might need to keep any one form variable, that would be:
1. Rename ```$_paymentProcessorObject``` to ```$_paymentProcessor```
2. Remove ```$_paymentProcessorObject```